### PR TITLE
Support for section ordering

### PR DESCRIPTION
**Pull request by khanchi97**
Closes https://github.com/coala/coala/issues/5038

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [ ] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by khanchi97** [comment]
@Makman2 can you please take a look at this **Section Ordering** PR :)

File: coalib/coala_main.py [hunk 1]
```diff
@@ -153,6 +154,12 @@ def run_coala(console_printer=None,
         if not sections['cli'].get('disable_caching', False):
             cache = FileCache(None, os.getcwd(), flush_cache)
 
+        if len(targets) != 0:
```
Comment (Makman2) [inline coalib/coala_main.py]: Does `if targets` work?

File: coalib/coala_main.py [hunk 2]
```diff
@@ -153,6 +154,12 @@ def run_coala(console_printer=None,
...
+            new_sections = collections.OrderedDict()
```
Comment (Makman2) [inline coalib/coala_main.py]: Do above `from collections import OrderedDict`, than you don't have to reference the module here.

File: coalib/coala_main.py [hunk 3]
```diff
@@ -153,6 +154,12 @@ def run_coala(console_printer=None,
...
+            for section_name in targets:
+                new_sections[section_name] = sections[section_name]
+            sections = new_sections
```
Comment (Makman2) [inline coalib/coala_main.py]:
```
sections = OrderedDict((section_name, sections[section_name]) for section_name in targets)
```

File: tests/coalaCITest.py [hunk 4]
```diff
@@ -74,6 +74,39 @@ def test_find_no_issues(self, debug=False):
             self.assertEqual(retval, 0,
                              'coala must return zero when successful')
 
+    def test_seciton_ordering(self, debug=False):
```
Comment (Makman2) [inline tests/coalaCITest.py]: typo: `seciton` --> `section`

**Comment by Makman2** [comment]
unack 47696e7268e1e81ccf8719aa7d1946702a10a917

**Comment by khanchi97** [comment]
Done :wink:

**Comment by Makman2** [comment]
Commit message: I would add "add":
`Add support for section ordering`

Not sure if we should also use the tag "coala" so
`coala: Add support for section ordering`
But that's optional, at least add "add" :P

**Comment by Makman2** [comment]
ack c22693e9f86ca906fba617c262dc76ddb37c7187

**Comment by Makman2** [comment]
unack c22693e

**Comment by khanchi97** [comment]
I would prefer this one ;)
`coala: Add support for section ordering`

File: coalib/coala_main.py [hunk 5]
```diff
@@ -153,6 +154,11 @@ def run_coala(console_printer=None,
         if not sections['cli'].get('disable_caching', False):
             cache = FileCache(None, os.getcwd(), flush_cache)
 
+        if targets:
+            sections = OrderedDict(
+                (section_name,
+                 sections[section_name]) for section_name in targets)
```
Comment (Makman2) [inline coalib/coala_main.py]: Oh just thinking and optional:
```
OrderedDict(
    (section_name, sections[section_name])
    for section_name in targets)
```

Comment (khanchi97) [reply coalib/coala_main.py]: okay, but should be last one :wink:

**Comment by Makman2** [comment]
ack 5ad32825566fd8ab95bff613679fdc538054d882

**Comment by Makman2** [comment]
@gitmate-bot rebase

**Comment by Makman2** [comment]
@gitmate-bot ff

**Pull request merged**